overlord/configstate/configcore/homedirs_test.go

```diff
@@ -0,0 +1,216 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore_test
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/configstate/configcore"
+)
+
+type mockedFileInfo struct {
+	isDir bool
+}
+
+func (f *mockedFileInfo) Name() string {
+	return ""
+}
+
+func (f *mockedFileInfo) Size() int64 {
+	return 0
+}
+
+func (f *mockedFileInfo) Mode() os.FileMode {
+	return 0
+}
+
+func (f *mockedFileInfo) ModTime() time.Time {
+	return time.Unix(0, 0)
+}
+
+func (f *mockedFileInfo) IsDir() bool {
+	return f.isDir
+}
+
+func (f *mockedFileInfo) Sys() interface{} {
+	return nil
+}
+
+type homedirsSuite struct {
+	configcoreSuite
+}
+
+var _ = Suite(&homedirsSuite{})
+
+func (s *homedirsSuite) SetUpTest(c *C) {
+	s.configcoreSuite.SetUpTest(c)
+
+	etcDir := filepath.Join(dirs.GlobalRootDir, "/etc/")
+	err := os.MkdirAll(etcDir, 0755)
+	c.Assert(err, IsNil)
+	s.AddCleanup(func() {
+		err := os.RemoveAll(etcDir)
+		c.Assert(err, IsNil)
+	})
+
+	restore := configcore.MockStat(func(path string) (os.FileInfo, error) {
+		if strings.HasPrefix(path, "/home/existingDir") {
+			return &mockedFileInfo{isDir: true}, nil
+		} else if strings.HasPrefix(path, "/home/existingFile") {
+			return &mockedFileInfo{isDir: false}, nil
+		} else {
+			return nil, errors.New("stat failed")
+		}
+	})
+	s.AddCleanup(restore)
+
+}
+
+func (s *homedirsSuite) TestValidationUnhappy(c *C) {
+	for _, testData := range []struct {
+		homedirs      string
+		expectedError string
+	}{
+		{"./here", `path "\./here" is not absolute`},
+		// empty path in list
+		{",/home", `path "" is not absolute`},
```

Comment:
(nitpick) maybe one test with a valid and an invalid pattern?

Reply:
This function is only testing error cases (and there's a test for an invalid pattern starting with "`/lib`" below). The only valid pattern is `/home`, and it's tested in the happy cases already.

Reply:
Sorry, I think I expressed myself badly, I was thinking of having a list where there is one good item and then one bad item. It's a bit silly (hence nitpick) but it ensures we actually split the list and look at each item (sort of, at least for the case where there are two elements etc). Something like: `{"/home/foo","[invalid]"}`. But again - probably a bit silly.

Reply:
One "fun" side effect - we cannot have directories with `,` in them if we use that as the separator. But maybe it's already excluded due to apparmor. It's also a bit silly have (home) dirs with that but I wanted to mention it (but probably not a big deal anyway).